### ISSUE TITLE
Change the way challenges are added

### ISSUE THREAD

**Issue by sublimemarch** [ask-again-later/character-system#35]
Under Challenges, there should be a button for "Add Creature Challenge" and a button for "Add Challenge." These open the standard modal we've used elsewhere. If you try to add more than two challenges, an alert pops up saying you can only have a maximum of two challenges.

The list of challenges that have been added should stay the same, with the icons for edit and delete.

**Comment by carlydotwebsite** [comment]
re: this, I'm working on it, and I feel like having [add creature challenge] [add challenge] next to each other just ends up being kind of confusing; maybe "add new challenge" for the second one? I feel like "new" fixes the problems with "custom".

**Comment by carlydotwebsite** [comment]
oh, and also, do we just want to just warn if there's a third challenge being added, or do we want to have it prevent them from actually adding the third? I was thinking we could just add validation messages on the page.

**Comment by sublimemarch** [comment]
Yup, I like the idea of "creature" and "new". Also, let's make sure that these are styled like secondary buttons, so we don't have a bunch of calls to action on the page. Actually... all of the buttons on this page that aren't the submit one should be secondary buttons. I'll make a new issue for that.

Yes, let's let them add several, but show validation messages.

**Comment by carlydotwebsite** [comment]
ok, neat. I have validation messages started in issue #37, so let's handle that in that branch instead, I guess? And I'll put a .button-secondary on all of these buttons so that they can all be styled at once easily.

**Comment by sublimemarch** [comment]
I've put the .button-secondary stuff in issue #45 

If the validation messages and secondary buttons are covered in other issues, do we still need this one? Does this capture anything that isn't in the other issues?

**Comment by carlydotwebsite** [comment]
This issue does still cover some items, such as splitting the challenge addition process on character sheets into two parts ("add creature challenge" and "new challenge") and doing some backend filtering stuff toward that end. I also added a checkbox to new challenges that checks whether it counts as a creature challenge, which today's meeting I thought indicated might be useful (e.g. for stuff like a "Siren" challenge, or misc. cryptid stuff)

I've submitted a pull request for review!
